pkg/build/registry/buildconfig/strategy.go

```diff
@@ -39,16 +52,17 @@ func (strategy) AllowUnconditionalUpdate() bool {
 	return false
 }
 
-// PrepareForCreate clears fields that are not allowed to be set by end users on creation.
-func (strategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
-	bc := obj.(*api.BuildConfig)
-	dropUnknownTriggers(bc)
-}
-
 // Canonicalize normalizes the object after validation.
 func (strategy) Canonicalize(obj runtime.Object) {
 }
 
+// PrepareForCreateCommon clears fields that are not allowed to be set by end users on creation.
+// This is shared by the Group and Legacy strategies.
+func (s strategy) PrepareForCreateCommon(ctx apirequest.Context, obj runtime.Object) {
```

Comment:
This could just be `PrepareForCreate` and the impl are just specific when they call, right?